js/reference.js

```diff
@@ -1669,13 +1669,33 @@
                 var column, keyColumns, referenceColumn;
 
                 // add column name into list of column projections if not in column projections set and data has changed
-                var addProjection = function(colName) {
+                var addProjection = function(colName, colType) {
                     // don't add a column name in if it's already there
                     // this can be the case for multi-edit
                     // and if the data is unchanged, no need to add the column name to the projections list
                     // NOTE: This doesn't properly verify if date/timestamp/timestamptz values were changed
-                    if ( (columnProjections.indexOf(colName) === -1) && (oldData[colName] != newData[colName]) ) {
-                        columnProjections.push(colName);
+                    if (columnProjections.indexOf(colName) === -1) {
+                        var typename = colType.rootName;
+                        var compareWithMoment = typename === 'date' || typename === 'timestamp' || typename === 'timestamptz';
+                        if (compareWithMoment) {
+                            var moment = module._moment;
+                            var formats = module._dataFormats;
+
+                            // default to DATE format, if timestamp or timestamptz, change the format used
+                            var formatToUse = formats.DATE;
+                            if (typename === 'timestamp') {
+                                formatToUse = formats.TIMESTAMP;
```

Comment:
I don't think this new format that you've added is needed.

You can use the same `YYYY-MM-DDTHH:mm:ssZ` format for both timestamp and timestamptz as long as you don't pass `true` for the last prop. The `true` prop is for "strict mode" which means that it will also validate the value. Which I don't think we need to do here.